feat(ui): Added support to add external URLs to project

smrutis1 opened the pull request:
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

* Removed "homepage", "wiki" and migrated it to "externalUrls" in project structure.
* Users now can add any number of urls(for e.g. "clearing" URL) in project through newly added "External Urls" section in UI.
* REST: Have kept the response containing "homepage" & "wiki" while getting projects with all details as it is.
  Have kept the newly added "exteralUrls" not readable and writable from REST for now as it was there for
  "homepage" URL and "wiki" URL.

> * Which issue is this pull request belonging to and how is it solving it? (*#1255*)
> * Did you add or update any new dependencies that are required for your change? - No

Issue: Closes #1255 

### Suggest Reviewer
> You can suggest reviewers here with an @mention.

### How To Test?
> How should these changes be tested by the reviewer?

* Run the migration scripts to verify migration of existing "homepage" and "wiki" to "externalUrls" and deletion of fields("wiki" and "homepage") after migration.
* Need to verify adding, editing, deleting external urls in the UI section of the project. 

> Have you implemented any additional tests? - No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

ravi110336 commented:
Tested the Migration scripts, works fine it listed out all the affected projects.
![image](https://user-images.githubusercontent.com/56516845/123223433-84306080-d4ee-11eb-90b4-99af708c76f1.png)

- External Url's for wiki, HomePage and Clearing State can be a multi-value field, it allows to update the project with multiple wiki urls, but when we store in the DB it has only the latest wiki url and same gets display in the UI.

![image](https://user-images.githubusercontent.com/56516845/123223840-dc676280-d4ee-11eb-8dc0-5c79e417b03d.png)

- Tried with the wrong wiki url and try to update the project, it doesn't allow to update the project but no alert to user like "The entered URL is incorrect", may be we can send some alert/warning if the url is incorrect.

![image](https://user-images.githubusercontent.com/56516845/123224660-a7a7db00-d4ef-11eb-980d-f10c0745bf66.png)

smrutis1 commented:
Thanks for review and test.

> * External Url's for wiki, HomePage and Clearing State can be a multi-value field, it allows to update the project with multiple wiki urls, but when we store in the DB it has only the latest wiki url and same gets display in the UI.
> 

Added support to add multiple URLs with the same name.


> * Tried with the wrong wiki url and try to update the project, it doesn't allow to update the project but no alert to user like "The entered URL is incorrect", may be we can send some alert/warning if the url is incorrect.

Added the error message. 

![image](https://user-images.githubusercontent.com/46560890/123370532-869dc380-d59d-11eb-965f-5ce5415ae6d3.png)

Note: The URL should start with `http:` or `https:`. Validation done using `<input type="URL" ...`

ravi110336 commented:
Thanks @smrutis1 for the changes, reviewed and tested the PR and working fine from the UI and REST.

![Screenshot from 2021-06-28 10-41-17](https://user-images.githubusercontent.com/56516845/123583760-ccf15d80-d7fd-11eb-80a4-132a89f00cc4.png)
